helm-charts/basehub/templates/home-space-reporter.yaml

```diff
@@ -1,57 +1,67 @@
-# Deploy a prometheus node_exporter with the same home directory
-# we have for our hub mounted so we can monitor free space usage.
 {{- if or .Values.nfs.enabled .Values.azureFile.enabled }}
+# To provide data for the jupyterhub/grafana-dashboards dashboard about free
+# space in the shared volume, which contains users home folders and the shared
+# folder, we deploy prometheus node-exporter to collect this data for prometheus
+# server to scrape.
+#
+# This is based on the Deployment manifest in jupyterhub/grafana-dashboards'
+# readme: https://github.com/jupyterhub/grafana-dashboards#additional-collectors
+#
 apiVersion: apps/v1
 kind: Deployment
 metadata:
+  name: shared-volume-metrics
   labels:
     app: jupyterhub
-    component: home-metrics
-  name: home-metrics
+    component: shared-volume-metrics
 spec:
   replicas: 1
   selector:
     matchLabels:
       app: jupyterhub
-      component: home-metrics
+      component: shared-volume-metrics
   template:
     metadata:
       annotations:
+        # This enables prometheus to actually scrape metrics from here
         prometheus.io/scrape: "true"
         prometheus.io/port: "9100"
       labels:
         app: jupyterhub
-        component: home-metrics
+        # This component label is used in our dashboard, so do not remove
+        component: shared-volume-metrics
     spec:
       containers:
-      - args:
-        # We only want filesystem stats
-        - --collector.disable-defaults
-        - --collector.filesystem
-        - --web.listen-address=:9100
-        image: quay.io/prometheus/node-exporter:v1.3.1
-        name: home-directory-exporter
-        ports:
-        - containerPort: 9100
-          name: metrics
-          protocol: TCP
-        securityContext:
-          allowPrivilegeEscalation: false
-        volumeMounts:
-          - name: home
-            mountPath: /home
-            # Mount it readonly to prevent accidental writes
-            readOnly: true
+        - name: shared-volume-exporter
+          image: quay.io/prometheus/node-exporter:v1.5.0
+          args:
+            # We only want filesystem stats
+            - --collector.disable-defaults
+            - --collector.filesystem
+            - --web.listen-address=:9100
+          ports:
+            - containerPort: 9100
+              name: metrics
+              protocol: TCP
+          securityContext:
+            allowPrivilegeEscalation: false
+            runAsGroup: 65534
+            runAsNonRoot: true
+            runAsUser: 65534
```

Comment:
@consideRatio can you please share the motivation behind moving these security context settings from a deployment level, to a container one? I understand what this does, but I'm curious to understand if this part of a "best practices" or has a some other motivation?

Reply:
Unless we have side-car or init containers added, this is equivalent. If we do, we didn't add them, so then its better if they aren't influenced by a podSecurityPolicy I think.

I've seen a pod's securityPolicy break assumptions for istio side-car containers added automatically in the past.

I made a comment in the commit adding this to Yuvi's PR: https://github.com/jupyterhub/grafana-dashboards/pull/51/commits/9db9ca8c4a15d0a20697eecad4e5d020c552e6e0

---

Note that this is a copy-paste mostly from the grafana-dashboards, so we are more in sync with whats documented there.

Reply:
Makes more sense now. Thanks @consideRatio